# EA1:

*By Daniel Deutsch and José Lucas Barretto*

In [13]:
import numpy as np
import pandas as pd
import statsmodels.tsa.stattools as sm

In [55]:
# read AAA corporate bonds data
df_corpbonds = pd.read_csv('datasets/AAA_corpbonds_yields.csv', na_values='.')

# convert date columns to datetime
df_corpbonds['DATE'] = pd.to_datetime(df_corpbonds['DATE'], format='%Y-%m-%d')

# apply log transform to values
df_corpbonds['DAAA'] = np.log(df_corpbonds['DAAA'])

In [284]:
df_corpbonds.dropna(inplace=True)

adf_res = sm.adfuller(df_corpbonds['DAAA'], regression='ct', maxlag=1, autolag=None, regresults=True)

In [292]:
print(adf_res[3].resols.summary())

                            OLS Regression Results                            
Dep. Variable:                      y   R-squared:                       0.007
Model:                            OLS   Adj. R-squared:                  0.005
Method:                 Least Squares   F-statistic:                     3.023
Date:                Wed, 08 Sep 2021   Prob (F-statistic):             0.0288
Time:                        16:09:56   Log-Likelihood:                 3346.8
No. Observations:                1249   AIC:                            -6686.
Df Residuals:                    1245   BIC:                            -6665.
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
x1            -0.0114      0.004     -2.926      0.0

1. Check if x1's t-stat is below critical value -> t-stat for x1 = -2.926 -> we do not reject $H_0: \rho = 0$.

2. Check if coefficient of the trend is statistically significant -> t-stat for x3 = -2.749 -> the coef. of the trend is not significant.

3. Run test again without trend.